MetadataMigration/README.md

```diff
@@ -0,0 +1,296 @@
+# Metadata Migration
+
+When performing a migration of a search cluster the metadata items such as indexes, templates, configuration and processes need to be in place before document data can be moved.  The metadata migration tool provides insight into what can be moved, if there are any issues moving that data to the target cluster and can deploy those changes.  By inspecting and analyzing the metadata issues can be discovered early in the overall migration timeline. 
+
+## Migration Lifecycle
+
+The lifecycle of a migration revolves around a set of clients that point to the source cluster, and at the end of the lifecycle they have been updated to point to a new target cluster.  There are many individual stages to this process, for this case we are focusing on the beginning of the migration where gathering information about what is going to be migrated and indicating the change of success.
+
+
+### Metadata migration lifecycle
+
+To handle moving more than data OpenSearch Migration includes we are tooling to inspect an existing cluster, generate a recommended configuration, and apply the migration onto a target cluster.
+
+```mermaid
+graph LR
+   subgraph "Metadata migration tool"
+      A[Evaluate] --> B[Configure]
+      B --> A
+      A --> |No issues found| D[Migrate]
+    end
+    D --> E[Verification]
+    E -.-> F([Start Data Migration])
+```
+
+### Evaluate
+By inspecting the source cluster and the target cluster, the metadata tooling will determine what items can be processed and any potential issues with them that would prevent a successful metadata migration.
+
+### Configure
+By inspecting the source and the items it contains some options can be automatically configured.  As some items require manually intervention to what metadata items should be migrated or how to resolve different kinds of conflicts updating the configuration narrows down this scope.
+
+### Migrate
+Following the configuration used in the evaluation the target cluster is updated with all of the metadata items. During the deployment there might have been issues these can be better understood by inspecting the output and/or detailed logs.
+
+### Verification
+Before starting data migration the target cluster can be inspected by running queries to confirm the state or any other checks before the longer process of migrating data from the source to the target cluster.
+
+## Walkthrough
```

Comment:
How is the tool installed? Need to figure out how to distribute / consume.

Possible options:
- (Default) Ship jar, include details on how to get a copy from opensearch.org website.
- Docker image, we already build docker images, we could publish the metadata image [1]

---
- [1] Turns out that OpenSearch-Benchmark already does this  https://opensearch.org/docs/latest/benchmark/user-guide/installing-benchmark/

Reply:
I like the idea to do a docker image. We could include a (possibly stripped down?) version of the console library too, and then we don't really have to worry about two versions of an interface.

Reply:
I've created a Jira task to track this https://opensearch.atlassian.net/browse/MIGRATIONS-1851